x-pack/plugins/security_solution/cypress/integration/detection_rules/bulk_edit_rules.spec.ts

```diff
@@ -0,0 +1,220 @@
+  waitForMixedRulesBulkEditModal,
+  openBulkEditAddTagsForm,
+  openBulkEditDeleteTagsForm,
+  typeTags,
+} from '../../tasks/rules_bulk_edit';
+
+import { hasIndexPatterns } from '../../tasks/rule_details';
+import { loginAndWaitForPageWithoutDateRange } from '../../tasks/login';
+
+import { SECURITY_DETECTIONS_RULES_URL } from '../../urls/navigation';
+import { createCustomRule } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import {
+  getExistingRule,
+  getNewOverrideRule,
+  getNewRule,
+  getNewThresholdRule,
+  totalNumberOfPrebuiltRules,
+} from '../../objects/rule';
+
+const RULE_NAME = 'Custom rule for bulk actions';
+
+const CUSTOM_INDEX_PATTERN_1 = 'custom-cypress-test-*';
+const DEFAULT_INDEX_PATTERNS = ['index-1-*', 'index-2-*'];
+const TAGS = ['cypress-tag-1', 'cypress-tag-2'];
+const OVERWRITE_INDEX_PATTERNS = ['overwrite-index-1-*', 'overwrite-index-2-*'];
+
+const customRule = {
+  ...getNewRule(),
+  index: DEFAULT_INDEX_PATTERNS,
+  name: RULE_NAME,
+};
+
+describe('Detection rules, bulk edit', () => {
+  beforeEach(() => {
+    cleanKibana();
+
+    createCustomRule(customRule, '1');
+    createCustomRule(getExistingRule(), '2');
+    createCustomRule(getNewOverrideRule(), '3');
+    createCustomRule(getNewThresholdRule(), '4');
+    createCustomRule({ ...getNewRule(), name: 'rule # 5' }, '5');
+    createCustomRule({ ...getNewRule(), name: 'rule # 6' }, '6');
+
+    loginAndWaitForPageWithoutDateRange(SECURITY_DETECTIONS_RULES_URL);
+    waitForRulesTableToBeLoaded();
+  });
+
+  it('should show modal windows when Elastic rules selected and edit only custom rules', () => {
+    cy.get(LOAD_PREBUILT_RULES_ON_PAGE_HEADER_BTN)
+      .pipe(($el) => $el.trigger('click'))
+      .should('not.exist');
+
+    // select few Elastic rules, check if we can't proceed further, as ELastic rules are not editable
+    // filter rules, only Elastic rule to show
+    cy.get(ELASTIC_RULES_BTN).click();
+    waitForRulesTableToBeRefreshed();
+
+    // check modal window for few selected rules
+    selectNumberOfRules(5);
+    clickAddIndexPatternsMenuItem();
+    waitForElasticRulesBulkEditModal(5);
+    cy.get(MODAL_CONFIRMATION_BTN).click();
+
+    // Select Elastic rules and custom rules, check mixed rules warning modal window, proceed with editing custom rules
+    cy.get(ELASTIC_RULES_BTN).click();
+    selectAllRules();
+    clickAddIndexPatternsMenuItem();
+    waitForMixedRulesBulkEditModal(totalNumberOfPrebuiltRules, 6);
+    cy.get(MODAL_CONFIRMATION_BTN).should('have.text', 'Edit custom rules').click();
+
+    typeIndexPatterns([CUSTOM_INDEX_PATTERN_1]);
+    confirmBulkEditForm();
+
+    // check if rule has been updated
+    cy.get(CUSTOM_RULES_BTN).click();
+    goToTheRuleDetailsOf(RULE_NAME);
+    hasIndexPatterns([...DEFAULT_INDEX_PATTERNS, CUSTOM_INDEX_PATTERN_1].join(''));
+  });
+
+  it('should add/delete/overwrite index patterns in rules', () => {
```

Comment:
Each test should check just one behaviour Here we are checking the addition, deletion and overwrite in the same. Please split it, in this way is going to be easier to understand what is exactly going wrong at first glance at the CI if one of the behaviours fails.

Reply:
Writing many tiny tests are expensive in Cypress, just overall environment setup takes more time than test itself. Following https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion I've decided to organise test as it in current state. It tests similar functionality, and in case of failure it's easy to find where it happened